### PR TITLE
[Auditbeat] Namespace system module to system.audit

### PR DESCRIPTION
**Pull request by cwurm**
So far, the system module has been using the top-level `system` object to store many of its fields (e.g. `system.user.group.name`), setting it up for potential clashes with the Metricbeat system module, which occupies the same top-level object.

This change namespaces all metricsets to the `system.audit` object (e.g. now `system.audit.user.group.name`), thus avoiding any potential field clashes - now and in the future.

Auditbeat and Metricbeat are still using the same module and metricset names in their configurations.

### CODE REVIEW

File: x-pack/auditbeat/module/system/host/host.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ import (
 const (
 	moduleName    = "system"
 	metricsetName = "host"
+	namespace     = "system.audit.host"
```
Comment (ruflin) [inline x-pack/auditbeat/module/system/host/host.go]: My concern is that even with https://github.com/elastic/beats/pull/8941 merged the dataset name here will still be `system.host` but should be `system.audit.host`. Do you want to keep it `host` as the name for the config?

Comment (cwurm) [reply x-pack/auditbeat/module/system/host/host.go]: Maybe we need a way to specify a custom dataset name? `{module}.{metricset}` will be weird for Auditbeat:

- auditd will be `auditd.auditd`
- file_integrity will be `file_integrity.file`

Comment (tsg) [reply x-pack/auditbeat/module/system/host/host.go]: +1. I'd say we merge this PR and make another one for the dataset?

Comment (ruflin) [reply x-pack/auditbeat/module/system/host/host.go]: SGTM

**Pull request merged**